UOP1_Project/Assets/Scripts/Sound/SoundEmitter.cs

```diff
@@ -24,13 +24,29 @@ void Start()
 		_audioSource.playOnAwake = false;
     }
 
+	public void OnRequest()
+	{
+		gameObject.SetActive(true);
+	}
+
+	public void OnReturn(Action onReturned)
+	{
+		StopSound();
+		gameObject.SetActive(false);
+	}
```

Comment:
You need to call `onReturned.Invoke();` after you complete the reset of your object. This is why objects are not being returned to the pool. It's a callback to allow for resets that must be completed over several frames. Take a look at the Example scene and check out PoolableParticle for an example implementation. For a graceful return of a SoundEmitter, you likely want to stop looping if it's looping, wait for the current playback to be finished, Stop(), and then return.

I'm currently working on documentation for Factory and Pool. This is helpful. I'll be sure to emphasize the importance of calling the callback. Or maybe we can think of something more foolproof?